app/scripts/modules/core/src/managed/Environments.tsx

```diff
@@ -1,5 +1,59 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 
-export default function Environments() {
-  return <div>For there shall be no greater pursuit than that towards desired state.</div>;
+import { Application, ApplicationDataSource } from '../application';
+import { IManagedApplicationSummary } from '../domain/IManagedEntity';
+import { ColumnHeader } from '../presentation/layout/ColumnHeader';
+import { ArtifactsList } from './ArtifactsList';
+import { EnvironmentsList } from './EnvironmentsList';
+
+import styles from './Environments.module.css';
+import { isEqual } from 'lodash';
+
+const CONTENT_WIDTH = 1200;
+
+export interface ISelectedArtifact {
+  name: string;
+  version: string;
+}
+
+interface IEnvironmentsProps {
+  app: Application;
+}
+
+export function Environments(props: IEnvironmentsProps) {
+  const { app } = props;
+  const dataSource: ApplicationDataSource<IManagedApplicationSummary<
+    'resources' | 'artifacts' | 'environments'
+  >> = app.getDataSource('environments');
+  const [selectedArtifact, setSelectedArtifact] = useState<ISelectedArtifact>();
+  const [environments, setEnvironments] = useState(dataSource.data);
+  const [isFiltersOpen] = useState(false);
```

Comment:
WIP?

Reply:
this whole thing is WIP 😄

Reply:
Yes, we'll eventually add this so I figured i'd just leave it in.